jupyterlab_gcedetails/auth_token_server/app.js

```diff
@@ -0,0 +1,73 @@
+// [START gae_node_request_example]
+const fs = require('fs');
+const path = require('path');
+const express = require('express');
+const cors = require('cors');
+const { google } = require('googleapis');
+
+const app = express();
+app.use(cors());
+
+/**
+ * To use OAuth2 authentication, we need access to a a CLIENT_ID, CLIENT_SECRET, AND REDIRECT_URI.  To get these credentials for your application, visit https://console.cloud.google.com/apis/credentials.
+ */
+const keyPath = path.join(__dirname, 'oauth2.keys.secrets.json');
+let keys = { redirect_uris: [''] };
+if (fs.existsSync(keyPath)) {
+  keys = require(keyPath).web;
+}
+
+/**
+ * Create a new OAuth2 client with the configured keys.
+ */
+const oauth2Client = new google.auth.OAuth2(
+  keys.client_id,
+  keys.client_secret,
+  keys.redirect_uris[0]
+);
+
+const scopes = ['https://www.googleapis.com/auth/compute'];
+
+google.options({ auth: oauth2Client });
+
+const errorFilePath = path.join(__dirname, 'error.html');
```

Comment:
Some rudimentary error handling, is there a specific way that we should do it? It's just kind of weird since we're not using typical http error codes